constraint/pkg/client/client.go

```diff
@@ -135,29 +166,72 @@ func (c *Client) AddTemplate(ctx context.Context, templ *templates.ConstraintTem
 		return resp, err
 	}
 
-	if err := c.driver.AddTemplate(ctx, templ); err != nil {
+	newDriverN := c.driverForTemplate(templ)
+
+	driver, ok := c.drivers[newDriverN]
+	if !ok {
+		return resp, fmt.Errorf("%w: available drivers: %v, wanted %q", clienterrors.ErrNoDriver, c.driverPriority, c.driverForTemplate(templ))
+	}
+
+	// TODO: because different targets may have different code sets,
```

Comment:
highligting TODO. you mentioned this is used for shared cache, since this is out of scope for CEL at this time, can we create an issue for this?

Reply:
This is a different TODO. This one is about supporting multi-target templates.

I probably wouldn't create a bug for this, since we are not actively working on multi-target templates ATM, and this would be part of that larger body of work. Having the TODO in the code is useful to avoid missing this nuance if/when we start working on it (issues are more likely to be missed).

Reply:
LMK if you'd prefer the issue though